flink-runtime/src/main/scala/org/apache/flink/runtime/minicluster/FlinkMiniCluster.scala

```diff
@@ -69,7 +69,7 @@ abstract class FlinkMiniCluster(
     ConfigConstants.JOB_MANAGER_IPC_ADDRESS_KEY,
     InetAddress.getByName("localhost").getHostAddress())
 
-  val configuration = generateConfiguration(userConfiguration)
+  protected val _configuration = generateConfiguration(userConfiguration)
```

Comment:
It's actually the original configuration which will usually have the job manager ipc port set to 0. Thus, in order to return a configuration which points to the current leader, we have to generate a new configuration via `configuration` which inserts the selected port. I'll rename the variable to `originalConfiguration`.